axiom/optimizer/RelationOp.cpp

```diff
@@ -215,6 +221,17 @@ std::string Cost::toString(bool /*detail*/, bool isUnit) const {
   return out.str();
 }
 
+void RelationOp::checkInputCardinality() const {
+  if (input_ != nullptr) {
+    const auto inputCardinality = input_->resultCardinality();
+    VELOX_CHECK(!std::isnan(inputCardinality));
+    VELOX_CHECK(std::isfinite(inputCardinality));
```

Comment:
```suggestion
    VELOX_CHECK(std::isfinite(inputCardinality));
```
isfinite check that number isn't nan or inf
https://en.cppreference.com/w/cpp/numeric/math/isfinite